src/Ryujinx/Ui/Windows/ControllerWindow.cs

```
@@ -923,6 +962,41 @@ private void HandleButtonPressed(ToggleButton button, bool forStick)
             inputThread.Start();
         }
 
+        private void HandleNextToggleButton(ToggleButton currentButton)
+        {
+            IEnumerable<ToggleButton> buttonOrder = GetCurrentButtonOrder();
+
+            ToggleButton nextButton = buttonOrder.SkipWhile(element => element != currentButton)
+                .Skip(1)
+                .FirstOrDefault();
+
+            nextButton?.Activate();
+        }
+
+        private IEnumerable<ToggleButton> GetCurrentButtonOrder()
+        {
+            ToggleButton[] baseOrder;
+            if (_inputDevice.ActiveId.StartsWith("keyboard"))
+            {
+                baseOrder = _buttonOrderKeyboard;
+            }
+            else if (_inputDevice.ActiveId.StartsWith("controller"))
+            {
+                baseOrder = _buttonOrderController;
+            }
+            else
+            {
+                throw new Exception("Input device not supported");
```

Can this just return an empty list?

---

It could, but returning an empty list would hide an error case.